src/uploader.ts

```diff
@@ -21,27 +27,48 @@ export default class Uploader {
   }
 
   private async uploadFile(entry: readdirp.EntryInfo) {
-    const readStream = fs.createReadStream(entry.fullPath);
     const destination = this.destination
       ? `${this.destination}/${entry.path}`
       : entry.path;
     info(
       `Deploying ${entry.path} by https://${this.storageEndpoint}/${this.storageName}/${destination}`
     );
+    let progressInterval: NodeJS.Timeout | undefined = undefined;
     return promiseRetry(
       async (attempt) => {
+        const totalBytes = (await fsPromises.stat(entry.fullPath)).size;
+        let lastUploadedBytes = 0;
+        let uploadedBytes = 0;
+
+        const readStream = fs.createReadStream(entry.fullPath);
+        readStream.on('data', (chunk) => {
+          uploadedBytes += chunk.length;
+        });
+        if (progressInterval) {
+          clearInterval(progressInterval);
+        }
+        progressInterval = setInterval(() => {
+          const uploadedMiB = formatMiB(uploadedBytes);
+          const totalMiB = formatMiB(totalBytes);
+          const speedMiB = formatMiB((uploadedBytes - lastUploadedBytes) / PROGRESS_REPORT_INTERVAL_SEC);
+          info(`Deploying ${entry.path}: ${uploadedMiB}/${totalMiB} MiB - ${speedMiB} MiB/s`);
```

Comment:
Also I'd try to change the output format here a bit. Currently it can shuffle around, so it isn't that readable. I'll try doing that once I have some more free time